### ISSUE TITLE
Configuring null foreign keys not to be mapped to corresponding object property

### ISSUE THREAD

**Issue by johnhamm** [agileobjects/AgileMapper#121]
Given I have four classes defined as:

```
    public class Person {
        public int Id { get; set; }
        public string Name { get; set; }
        public int? PetId { get; set; }
        public Pet Pet { get; set; }
    }

    public class PersonDto {
        public int Id { get; set; }
        public string Name { get; set; }
        public int? PetId { get; set; }
    }

    public class Pet {
        public int Id { get; set; }
        public string Name { get; set; }
    }

    public class PetDto {
        public int Id { get; set; }
        public string Name { get; set; }
    }
```

And I map a PersonDto to a Person:
```
    var personDto = new PersonDto { Id = 1, Name = "John" };
    var person = personDto.Map().ToANew<Person>();
```

The resulting object is:
```
{AgileMapperLab.Person}
  Id: 1
  Name: "John"
  PetId: null
  Pet: { }
    {AgileMapperLab.Pet}
      Id: 0
      Name: null
```
Even though PetId is null in the original PersonDto, AgileMapper creates a new empty Pet anyway.  The issue is that Entity Framework Core sees these as newly added objects and will add them to a database, even though pet is optional for a person.  

Is there a way to turn this behavior off in AgileMapper so that if a "foreign key" in an object is null (or 0), do not instantiate a new object for the matching object property?

**Comment by johnhamm** [comment]
I was able to bypass the new object creation by using IgnoreEntityKeys:

`personDto.Map().ToANew<Person>(cfg => cfg.IgnoreEntityKeys())`

Unfortunately, this also ignores the primary key Id field from the original DTO and Entity Framework Core then thinks it is a new object.

**Comment by SteveWilkes** [comment]
Hi John!

Hmm - that's certainly not the intended behaviour - I'll take a look and sort it out.

In the meantime, you can work around it using:

```C#
Mapper.WhenMapping
    .ToANew<Pet>()
    .If((o, p) => string.IsNullOrEmpty(p.Name))
    .MapToNull();
```

I've put together [a fiddle](https://dotnetfiddle.net/j6u2iU) showing this at work :)

Cheers,

Steve

**Comment by SteveWilkes** [comment]
This is fixed in the latest code -  if the only member of a complex type member mapped during a mapping is an identifier, and it isn't given a non-default value, the complex type member isn't populated. Release to follow.

Thanks for the feedback! :)

**Comment by SteveWilkes** [comment]
This is fixed as of [v1.3-preview2](https://www.nuget.org/packages/AgileObjects.AgileMapper/1.3.0-preview2).
